code/modules/antagonists/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,1093 @@
+#define EMPOWERED_THRALL_LIMIT 3
+
+/obj/effect/proc_holder/spell/proc/shadowling_check(var/mob/living/carbon/human/H)
+	if(!H || !istype(H)) return
+	if(H.dna && H.dna.species && H.dna.species.id == "shadowling" && IS_SHADOW(H)) return TRUE
+	if(H.dna && H.dna.species && H.dna.species.id == "l_shadowling" && IS_THRALL(H)) return TRUE
+
+	if(!IS_SHADOW_OR_THRALL(usr)) to_chat(usr, span_warning("You can't wrap your head around how to do this."))
+	else if(IS_THRALL(usr)) to_chat(usr, span_warning("You aren't powerful enough to do this."))
+	else if(IS_SHADOW(usr)) to_chat(usr, span_warning("Your telepathic ability is suppressed. Hatch or use Rapid Re-Hatch first."))
+	return FALSE
+	
+/obj/effect/proc_holder/spell/targeted/shadowling //Stuns and mutes a human target for 10 seconds
+	ranged_mousepointer = 'icons/effects/cult_target.dmi'
+	var/mob/living/user
+	var/mob/living/target
+
+/obj/effect/proc_holder/spell/targeted/shadowling/Click()
+	var/mob/living/user = usr
+	if(!istype(user))
+		return
+	var/msg
+	if(!can_cast(user))
+		msg = span_warning("You can no longer cast [name]!")
+		remove_ranged_ability(msg)
+		return
+	if(active)
+		remove_ranged_ability()
+	else
+		add_ranged_ability(user, null, TRUE)
+	if(action)
+		action.UpdateButtonIcon()
+
+/obj/effect/proc_holder/spell/targeted/shadowling/InterceptClickOn(mob/living/caller, params, atom/t)
+	if(!isliving(t))
+		to_chat(caller, span_warning("You may only use this ability on living things!"))
+		revert_cast()
+		return
+	user = caller
+	target = t
+	if(!shadowling_check(user))
+		revert_cast()
+		return
+
+/obj/effect/proc_holder/spell/targeted/shadowling/revert_cast()
+	. = ..()
+	remove_ranged_ability()
+
+/obj/effect/proc_holder/spell/targeted/shadowling/start_recharge()
+	. = ..()
+	if(action)
+		action.UpdateButtonIcon()
+
+/obj/effect/proc_holder/spell/targeted/shadowling/glare //Stuns and mutes a human target for 10 seconds
+	name = "Glare"
+	desc = "Disrupts the target's motor and speech abilities."
+	panel = "Shadowling Abilities"
+	charge_max = 300
+	human_req = TRUE
+	clothes_req = FALSE
+	action_icon_state = "glare"
+	action_icon = 'icons/mob/actions.dmi'
+
+/obj/effect/proc_holder/spell/targeted/shadowling/glare/InterceptClickOn(mob/living/caller, params, atom/t)
+	. = ..()
+	if(!target)
+		return
+	if(target.stat)
+		to_chat(usr, span_warning("[target] must be conscious!"))
+		revert_cast()
+		return
+	if(IS_SHADOW_OR_THRALL(target))
+		to_chat(usr, span_warning("You cannot glare at allies!"))
+		revert_cast()
+		return
+	var/mob/living/carbon/human/M = target
+	usr.visible_message(span_warning("<b>[usr]'s eyes flash a purpleish-red!"))
+	var/distance = get_dist(target, usr)
+	if (distance <= 2)
+		target.visible_message("<span class='danger'>[target] suddendly collapses...</span>")
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and you lose control of your movements!</span>")
```

Comment:
isAdjacent + spans

Reply:
I think it needs to have a two tile range instead of requiering the shadowling to be adjacent to it's target